src/logic/useTravelTime.ts

```diff
@@ -0,0 +1,74 @@
+import { useState, useEffect, useMemo, useCallback } from 'react'
+import { isEqual } from 'lodash'
+import service from '../service'
+import { TripPattern, Coordinates } from '@entur/sdk'
+import { useSettingsContext } from '../settings'
+import useStopPlacesWithDepartures from './useStopPlacesWithDepartures'
+import { StopPlaceWithDepartures } from '../types'
+import { usePrevious } from '../utils'
+
+async function fetchTravelTime(
+    stopPlaces: StopPlaceWithDepartures[],
+    from: Coordinates,
+): Promise<TripPattern[][]> {
+    const travelTimes = Promise.all(
+        stopPlaces.map(
+            async (stopPlace) =>
+                await service.getTripPatterns(
+                    {
+                        from: {
+                            name: 'pin',
+                            coordinates: from,
+                        },
+                        to: {
+                            name: stopPlace.name,
+                            place: stopPlace.id,
+                            coordinates: {
+                                latitude: stopPlace.latitude ?? 0,
+                                longitude: stopPlace.longitude ?? 0,
+                            },
+                        },
+                        modes: ['foot'],
+                    },
+                    undefined,
+                ),
+        ),
+    )
+    return travelTimes
+}
+
+export default function useTravelTime(): TripPattern[][] | null {
+    const [settings] = useSettingsContext()
+    const [travelTime, setTravelTime] = useState<TripPattern[][] | null>(null)
+
+    const {
+        latitude: fromLatitude,
+        longitude: fromLongitude,
+    } = settings?.coordinates ?? {
+        latitude: 0,
+        longitude: 0,
+    }
+
+    const stopPlacesWithDepartures = useStopPlacesWithDepartures()
```

Comment:
I staden for å bruke denne hooken her, som eg antar kan føre til doble nettverkskall når den blir også brukt i visningstypen, så kan du heller ta inn ei liste med stopPlaces som argument til denne hooken:

```
 export default function useTravelTime(stopPlaces: StopPlace[]): TripPattern[][] | null { 
```

Reply:
Også godt poeng, også endret!